scripts/compactor_runner.py

```diff
@@ -357,22 +329,118 @@ def _complete_config(self, args):
             config.startCheckpointing = args.startCheckpointing
         return config
 
+    def _configure_server_address(self, compactor_config):
+        """
+        Configure Server hostname and port from args and config file.
+        This method invokes resolve ip method to configure from network interfaces
+        when the hostname is not present in args or config file.
+
+        :param compactor_config: values parsed from config file
+        :return:
+        """
+        # configuring hostname and port
+        pod_name = os.environ.get(POD_NAME, "default_name")
+        pod_namespace = os.environ.get(POD_NAMESPACE, "default_namespace")
+        if pod_name != "default_name" and pod_namespace != "default_namespace":
+            self._config.hostname = pod_name + ".corfu-headless." + pod_namespace + ".svc.cluster.local"
+
+        server_address = compactor_config["ServerAddress"]
+        if not self._config.hostname:
+            if self._config.network_interface and self._config.network_interface_version:
+                self._print_and_log("WARNING: _configure_server_address: Couldn't load the hostname from args."
+                                    " Continuing self IP discovery with {}"
+                                    " interface and {} version from args.".format(self._config.network_interface,
+                                                                                  self._config.network_interface_version))
+                self._config.hostname = self._resolve_ip_address(
+                    network_interface=self._config.network_interface,
+                    network_interface_version=self._config.network_interface_version
+                )
+            elif server_address:
+                self._print_and_log("_configure_server_address: Found the server_address from config file: {}"
+                                    .format(server_address))
+                if server_address["Hostname"]:
+                    self._config.hostname = server_address["Hostname"]
+                # Do self ip discovery since hostname is not configured
+                elif server_address["NetworkInterfaceVersion"] \
+                        and server_address["NetworkInterface"]:
+                    self._print_and_log("WARNING: _configure_server_address: Couldn't load the hostname from args or"
```

Comment:
nit: Hostname not found in config file or passed through args

Reply:
Done, Thank you.